trust-quorum/src/coordinator_state.rs

```diff
@@ -137,39 +145,76 @@ impl CoordinatorState {
     // This method is "in progress" - allow unused parameters for now
     #[allow(unused)]
     pub fn send_msgs(&mut self, now: Instant, outbox: &mut Vec<Envelope>) {
-        match &self.op {
-            CoordinatorOperation::CollectShares {
-                epoch,
-                members,
-                collected_shares,
-                ..
-            } => {}
-            CoordinatorOperation::CollectLrtqShares { members, shares } => {}
-            CoordinatorOperation::Prepare { prepares, prepare_acks } => {
-                for (platform_id, prepare) in prepares.clone().into_iter() {
-                    outbox.push(Envelope {
-                        to: platform_id,
-                        from: self.reconfigure_msg.coordinator_id().clone(),
-                        msg: PeerMsg::Prepare(prepare),
-                    });
+        if now >= self.retry_deadline {
+            self.retry_deadline = now + self.reconfigure_msg.retry_timeout();
+            match &self.op {
+                CoordinatorOperation::CollectShares {
+                    epoch,
+                    members,
+                    collected_shares,
+                    ..
+                } => {}
+                CoordinatorOperation::CollectLrtqShares { members, shares } => {
+                }
+                CoordinatorOperation::Prepare { prepares, prepare_acks } => {
+                    for (platform_id, prepare) in prepares.clone().into_iter() {
+                        outbox.push(Envelope {
+                            to: platform_id,
+                            from: self.reconfigure_msg.coordinator_id().clone(),
+                            msg: PeerMsg::Prepare(prepare),
+                        });
+                    }
                 }
             }
         }
     }
+
+    pub fn ack_prepare(&mut self, from: PlatformId) {
+        match &mut self.op {
+            CoordinatorOperation::Prepare {
+                prepares, prepare_acks, ..
+            } => {
+                if !self.configuration.members.contains_key(&from) {
+                    warn!(
```

Comment:
Should this be `debug_assert` as well?

Reply:
No, I don't think that helps. These messages are sent from other replicas, and we don't want to crash on unexpected input. Yes it's probably a bug in the protocol implementation, but philosophically it is "user input" that could be issued by an attacker, and I think asserting, even in debug mode, does more harm than good.